### DISCUSSION TITLE
Simple example of how to remove an option from a list when clicked requested

### DISCUSSION THREAD

**Discussion by damienrj** [Textualize/textual#4456] [Q&A]
Hello, 
I have been learning textual and there are a lot of great examples of how to setup widgets like the option list, and using buttons but less clear how that extends to something like option list. But I have been trying to figure out how I could it with some inits but I think I am probably missing a better way.  So the base on the example from the docs https://textual.textualize.io/widgets/option_list/#options-as-option-instances, what would be a better way to have the same widget but  use the event of selecting an option make the option selected removed (was trying to do this as a self contained example)?

Below is the working solution I have currently.

```python
from textual.app import App, ComposeResult
from textual.widgets import Footer, Header, OptionList
from textual import on
from textual.widgets.option_list import Option, Separator

class OptionListApp(App[None]):
    # CSS_PATH = "option_list.tcss"
    
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.option_list = OptionList(
            Option("Aerilon", id="aer"),
            Option("Aquaria", id="aqu"),
            Option("Canceron", id="can"),
            Option("Caprica", id="cap"),
            Option("Gemenon", id="gem"),
            Option("Leonis", id="leo"),
            Option("Libran", id="lib"),
            Option("Picon", id="pic"),
            Option("Sagittaron", id="sag"),
            Option("Scorpia", id="sco"),
            Separator(),
            Option("Tauron", id="tau"),
            Separator(),
            Option("Virgon", id="vir"),
        )

    def compose(self) -> ComposeResult:
        yield Header()
        yield self.option_list
        yield Footer()

    @on(OptionList.OptionSelected)
    async def option_selected(self, event: OptionList.OptionSelected) -> None:
        self.option_list.remove_option_at_index(event.option_index)

if __name__ == "__main__":
    OptionListApp().run()
```

**Comment by davep** [comment]
It might help people if you clarified your question a bit; he subject and the main body don't seem to be related, and it's not really cler what the question is. It might be a good to ask one or more specific questions.

**Reply by damienrj** [reply]
updated with more context.
